OPHD/States/MapViewStateUi.cpp

```diff
@@ -115,10 +115,38 @@ void MapViewState::initUi()
 	// Initial Structures
 	mStructures.addItem(constants::SEED_LANDER, 0, StructureID::SID_SEED_LANDER);
 
+
+	// tooltip control sizes
+	constexpr auto hudHeight = constants::RESOURCE_ICON_SIZE + constants::MARGIN_TIGHT * 2;
+	mTooltipResourceBreakdown.size({ 265, hudHeight });
+
+	mTooltipResourceStorage.position({ 275, 0 });
+	mTooltipResourceStorage.size({ 85, hudHeight });
+
+	mTooltipFoodStorage.position({ 420, 0 });
+	mTooltipFoodStorage.size({ 75, hudHeight });
+
+	mTooltipEnergy.position({ 560, 0 });
+	mTooltipEnergy.size({ 55, hudHeight });
+
+	mTooltipPopulation.position({ 670, 0 });
+	mTooltipPopulation.size({ 75, hudHeight });
+
+	mTooltipCurrentTurns.size({ 45, hudHeight });
+	mTooltipSystemButton.size({ hudHeight, hudHeight });
```

Comment:
I'm somewhat concerned by this. It seems odd to be adding dummy controls, though I suspect this is because a lot of GUI rendering was done with raw draw calls, rather than creating controls to draw the UI.

I suppose this might be a sign to move some of that rendering away from raw draw calls and over to controls to draw those items.

Reply:
That's exactly what I was thinking. Since the ToolTip interface is implemented in terms of a Control, I needed to do this to add tooltips to the HUD.

Which is a huge, _ugly_ bit of code that could be done _way_ better using some derived Control type that would act in the intended way, perhaps organized into a tool-strip of sorts? Might be overthinking the design there but the basic idea is a 'bar' with a bunch of 'panels' in it with customized content. Then it would simply be a matter of having a ToolBar or whatever with a single call to update() it which would then update all of its 'children' and each of those 'children' could easily be added to the ToolTip. Each 'panel' object would override it's 'draw' function in some way for the customized content. Could maybe sub-in a lambda for it or just derive new types from 'panel' that provide their customized content.

I guess you could call this a placeholder until that's done. It's a semi-complex idea that's out of the context of this PR but definitely something I want to look into going forward.